File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let ans = 0;
    function dfs(node, sum){
        let s = sum;
        if(node === numbers.length){
           if(sum === target){
            ans++;
        }
            return;
        }
        s -= numbers[node];
        dfs(node + 1, s)  
        s += numbers[node] * 2;
        dfs(node + 1, s);
    }
    dfs(0,0);
    return ans;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    n = len(numbers)
    
    def dfs(total, depth):
        nonlocal answer
        if depth == n:
            if total == target:
                answer += 1
            return
        dfs(total + numbers[depth], depth + 1)
        dfs(total - numbers[depth], depth + 1)        
    dfs(0,0)
    return answer

